add stakeholder need (#1243)
spdx/spdx-3-model: model/Core/Vocabularies/RelationshipType.md

- hasEvidence: Every `to` Element is considered as evidence for the `from` Element (`from` hasEvidence `to`).
- hasExample: Every `to` Element is an example for the `from` Element (`from` hasExample `to`).
- hasHost: The `from` Build was run on the `to` Element during a LifecycleScopeType period (e.g. the host that the build runs on).

> - hasStakeholderNeed: the `from` Element has the  `to` StakeholderNeed.

Author: ```suggestion
- hasNeed: the `from` Agent has the `to` Need.
```

- hasEvidence: Every `to` Element is considered as evidence for the `from` Element (`from` hasEvidence `to`).
- hasExample: Every `to` Element is an example for the `from` Element (`from` hasExample `to`).
- hasHost: The `from` Build was run on the `to` Element during a LifecycleScopeType period (e.g. the host that the build runs on).

> - hasNeed: the `from` Agent has the `to` Need.

Reviewer: I think this is reflected in @stevenc-stb [comment here](https://github.com/spdx/spdx-3-model/pull/1243#issuecomment-4360531886), but just for the sake of completeness, I think we need to modify this.

```suggestion
- hasNeed: the `from` Role has the `to` Need.
```

Author: Done

> - hasNeed: the `from` Agent has the `to` Need.

Author: ```suggestion
- hasNeed: the `from` Role has the `to` Need.
```

- hasHost: The `from` Build was run on the `to` Element during a LifecycleScopeType period (e.g. the host that the build runs on).
- hasInput: The `from` Build, DefinedProcess or Action element has each `to` Element as an input.
- hasMetadata: Every `to` Element is metadata about the `from` Element (`from` hasMetadata `to`).

> - hasNeed: the `from` Role has the `to` Need.

Reviewer 2: ```suggestion
- hasNeed: The `from` Role has the `to` Need.
```

> - hasNeed: The `from` Role has the `to` /FunctionalSafety/Need.

Reviewer 3: I do not agree that a Role is the only thing that can have a "Need". Sometime you have a Need, that results from an assumption of another system component. It can also come from a contract. And in these cases you would have to make up a Role, just to satisfy this statement. I'd suggest to not use "Role", but "Element".

- hasVariant: Every `to` Element is a variant the `from` Element (`from` hasVariant `to`).
- implementedBy: The `from` Requirement is implemented in the `to` Element(s).
- invokedBy: The `from` Element was invoked by the `to` Agent, during a LifecycleScopeType period (for example, a Build element that describes a build step).
- isQualifiedFor: The `from` Agent is qualified for or can provided the `to` Role.

> - locatedAt: `from` element located at a specific `to` location. A time period is optional.

Reviewer 2: ```suggestion
- locatedAt: `from` Element located at a specific `to` Location. A time period is optional.
```

- modifiedBy: The `from` Element is modified by each `to` Element.
- other: Every `to` Element is related to the `from` Element where the relationship type is not described by any of the SPDX relationship types (this relationship is directionless).
- packagedBy: Every `to` Element is a packaged instance of the `from` Element (`from` packagedBy `to`).
- patchedBy: Every `to` Element is a patch for the `from` Element (`from` patchedBy `to`).

> - performedBy: Every `from` action is performedBy `to` Agent.

Reviewer 2: ```suggestion
- performedBy: Every `from` Action is performedBy `to` Agent.
```

- pretrainedOn: The `from` Element has been pretrained on the `to` Element(s).
- providesSupportFor: The `from` Agent provides support for each `to` Artifact. The use of the `providesSupportFor` type is constrained to `SupportRelationship` classed relationships.
- publishedBy: Designates a `from` Vulnerability was made available for public use or reference by each `to` Agent.
- reportedBy: Designates a `from` Vulnerability was first reported to a project, vendor, or tracking database for formal identification by each `to` Agent.
- republishedBy: Designates a `from` Vulnerability's details were tracked, aggregated, and/or enriched to improve context (i.e. NVD) by each `to` Agent.
- resolved: The `to` OutOfSpecAction is resolved in the `from` ResolutionAction.
- runsOn: The `from` Element (the instructions) of runs on each `to` Hardware (processing element), during a LifecycleScopeType period.

> - satisfies: The `from` Requirement satisfies `to` need. Note: The Requirement is not generally intended to singularly satisfy a particular need. It is more likely that a set of higher level requirements is required to achieve a reasonable satisfaction of an expressed need.

Reviewer 2: ```suggestion
- satisfies: The `from` Requirement satisfies `to` Need. Note: The Requirement is not generally intended to singularly satisfy a particular Need. It is more likely that a set of higher level Requirements is required to achieve a reasonable satisfaction of an expressed Need.
```

- reportedBy: Designates a `from` Vulnerability was first reported to a project, vendor, or tracking database for formal identification by each `to` Agent.
- republishedBy: Designates a `from` Vulnerability's details were tracked, aggregated, and/or enriched to improve context (i.e. NVD) by each `to` Agent.
- resolved: The `to` OutOfSpecAction is resolved in the `from` ResolutionAction.

> - satisfies: The `from` Requirement satisfies `to` need.

Author: ```suggestion
- satisfies: The `from` Requirement satisfies `to` need. Note: The Requirement is not intended to singularly satisfy a particular need. It is more likely than a set of higher level requirements.
```

> - satisfies: The `from` Requirement satisfies `to` need. 

Author: ```suggestion
- satisfies: The `from` Requirement satisfies `to` need.
```

> - satisfiesNeed: The `from` Requirement is help satisfies the `to` need. 

Author: ```suggestion
- satisfies: The `from` Requirement satisfies `to` need. 
```

> - satisfiesNeed: The `need` StakeholderNeed is Refined in the `to` Requirement.

Author: ```suggestion
- satisfiesNeed: The `from` Requirement is help satisfies the `to` need. 
```

> - requirementRefinesStakeholderNeed: The `from` StakeholderNeed is Refined in the `to` Requirement.

Reviewer: In a decomposed requirement structure, a higher level requirement can satisfy a need. This way we can also add other things like `satisfiesHazard` and `satisfiesThreat`.

```suggestion
- satisfiesNeed: The `need` StakeholderNeed is Refined in the `to` Requirement.
```

> - satisfies: The `from` Requirement satisfies `to` need. Note: The Requirement is not intended to singularly satisfy a particular need. It is more likely than a set of higher level requirements.

Reviewer: ```suggestion
- satisfies: The `from` Requirement satisfies `to` need. Note: The Requirement is not generally intended to singularly satisfy a particular need. It is more likely that a set of higher level requirements is required to achieve a reasonable satisfaction of an expressed need.
```

> - satisfies: The `from` Requirement satisfies `to` need. Note: The Requirement is not intended to singularly satisfy a particular need. It is more likely than a set of higher level requirements.

Reviewer 2: ```suggestion
- satisfies: The `from` Requirement satisfies `to` Need. Note: The Requirement is not intended to singularly satisfy a particular Need. It is more likely than a set of higher level requirements.
```

Author: i think this is fixed now.

- reportedBy: Designates a `from` Vulnerability was first reported to a project, vendor, or tracking database for formal identification by each `to` Agent.
- republishedBy: Designates a `from` Vulnerability's details were tracked, aggregated, and/or enriched to improve context (i.e. NVD) by each `to` Agent.
- resolved: The `to` OutOfSpecAction is resolved in the `from` ResolutionAction.

> - satisfies: The `from` Requirement satisfies `to` need. Note: The Requirement is not generally intended to singularly satisfy a particular need. It is more likely that a set of higher level requirements is required to achieve a reasonable satisfaction of an expressed need.

Reviewer 2: ```suggestion
- satisfies: The `from` Requirement satisfies `to` Need. Note: The Requirement is not generally intended to singularly satisfy a particular Need. It is more likely that a set of higher level requirements is required to achieve a reasonable satisfaction of an expressed Need.
```

Actually not entirely sure where should do the capitalized. Just flag.

- republishedBy: Designates a `from` /Security/Vulnerability's details were tracked, aggregated, and/or enriched to improve context (i.e. NVD) by each `to` Agent.
- resolved: The `to` /SupplyChain/OutOfSpecAction is resolved in the `from` /SupplyChain/ResolutionAction.
- runsOn: The `from` Element (the instructions) runs on each `to` /Hardware/Hardware (processing element), during a LifecycleScopeType period.

> - satisfies: The `from` Requirement satisfies `to` /FunctionalSafety/Need. Note: The Requirement is not generally intended to singularly satisfy a particular Need. It is more likely that a set of higher level Requirements is required to achieve a reasonable satisfaction of an expressed /FunctionalSafety/Need.

Reviewer 4: I'm thinking may be "satisfies" may be useful beyond requiement class.   Specification for example.
